x-pack/plugins/xpack_main/server/lib/xpack_info_license.js

```diff
@@ -65,4 +68,46 @@ export class XPackInfoLicense {
   getType() {
     return get(this._getRawLicense(), 'type');
   }
+
+  /**
+   * Returns mode of the license (basic, gold etc.). This is the "effective" type of the license.
+   * @returns {string|undefined}
+   */
+  getMode() {
```

Comment:
How is this different from type, above?

Reply:
TL;DR It's generally the same thing as `type`, but it's the one that you should check (same as `isOneOf` does above in this class).

`mode` is the operational mode that ES has enabled (the license `type`s we're used to). Technically it can be different from `type`, but in general they're the exact same value. This comes back from `GET /_xpack`, which `xpack_main` uses as opposed to `GET /_xpack/license`, which reports the mode / type as the same value.